web/js/modules/natural-events/util.js

```diff
@@ -27,3 +36,156 @@ export function serializeEvent(currentItemState) {
+  }
+
+  let showAll;
+  if (parameters.efs === 'true' || !parameters.efs) {
+    showAll = true;
+  } else if (parameters.efs === 'false') {
+    showAll = false;
+  }
+
+  return update(stateFromLocation, {
+    events: {
+      selectedCategories: {
+        $set: selectedCategories,
+      },
+      selectedDates: {
+        start: { $set: selectedStartDate },
+        end: { $set: selectedEndDate },
+      },
+      showAll: {
+        $set: showAll,
+      },
+    },
+  });
+}
+
+export function getEventsRequestURL (state) {
+  const {
+    config, proj, events, map,
+  } = state;
+  const {
+    selectedCategories,
+    selectedDates,
+    showAll,
+  } = events;
+  const baseUrl = lodashGet(config, 'features.naturalEvents.host');
+  const mockEvents = lodashGet(config, 'parameters.mockEvents');
+  if (mockEvents) {
+    // eslint-disable-next-line no-console
+    console.warn(`Using mock events data: ${mockEvents}`);
+    return mockEvents === 'true'
+      ? 'mock/events_data.json'
+      : `mock/events_data.json-${mockEvents}`;
+  }
+  // Rough extents used for EONET projection extent filtering.  Polar extents do not
+  // represent exact bounds seen in app since they are expressed in EPSG:4326
+  // format which is the only format the API supports
+  const extentBounds = {
+    'EPSG:4326': [-180, 90, 180, -90],
+    'EPSG:3413': [-180, 40, 180, 90],
+    'EPSG:3031': [-180, -90, 180, -40],
+  };
+  const { crs } = proj.selected;
+  const selectedMap = map && map.ui.selected;
+  const bbox = !showAll && selectedMap && selectedMap.getView().calculateExtent();
+  const { start, end } = selectedDates;
+  const params = {
+    status: 'all',
+    limit: LIMIT_EVENT_REQUEST_COUNT,
+  };
+  const useBbox = bbox && bbox.length && crs === 'EPSG:4326';
+  params.bbox = useBbox ? bbox : extentBounds[crs];
+
+  if (start && end) {
+    params.start = moment.utc(start).format('YYYY-MM-DD');
+    params.end = moment.utc(end).format('YYYY-MM-DD');
+  }
+  if (selectedCategories.length) {
+    params.category = selectedCategories.map(({ id }) => id).join(',');
+  }
+  return `${baseUrl}/events${util.toQueryString(params)}`;
+}
+
+/**
+ *
+ * @param {*} event
+ */
+export function getDefaultEventDate(event) {
+  const preDate = event.geometry[0] && event.geometry[0].date;
+  let date = new Date(preDate).toISOString().split('T')[0];
+  if (event.geometry.length < 2) return date;
+  const category = event.categories.title || event.categories[0].title;
+  const today = new Date().toISOString().split('T')[0];
+  // For storms that happened today, get previous date
+  if (date === today && category === 'Severe Storms') {
+    [date] = new Date(event.geometry[1].date).toISOString().split('T');
+  }
+  return date;
+}
+
+/**
+ * Validate whether an event and all it's points/polygons fall within
+ * the provded projection's maxExtent
+ *
+ * @param {*} event
+ * @param {*} proj
+ * @returns
+ */
+export const validateEventCoords = (event, proj) => {
```

Comment:
I got an error after selecting a multi-date storm event in Geographic projection and then switching to Arctic:
![image](https://user-images.githubusercontent.com/24417194/123169403-41677d80-d447-11eb-81fc-cd79ae7a7f6f.png)
http://localhost:3000/?v=129.1017578125,7.2435546875,152.4982421875,21.9564453125&e=EONET_5423,2021-06-22&efs=true&efd=2021-02-23,2021-06-23&efc=dustHaze,manmade,seaLakeIce,severeStorms,snow,volcanoes,waterColor,wildfires&l=IMERG_Precipitation_Rate,VIIRS_SNPP_DayNightBand_ENCC(hidden),VIIRS_SNPP_DayNightBand_At_Sensor_Radiance(hidden),Reference_Labels_15m,Reference_Features_15m,Coastlines_15m(hidden),VIIRS_NOAA20_CorrectedReflectance_TrueColor(hidden),VIIRS_SNPP_CorrectedReflectance_TrueColor,MODIS_Aqua_CorrectedReflectance_TrueColor(hidden),MODIS_Terra_CorrectedReflectance_TrueColor&lg=true&t=2021-06-22-T00%3A00%3A00Z

Reply:
Good catch, should be fixed now (after next push of updates)